### ISSUE TITLE
Handle optional type in RxKotlin2?

### ISSUE THREAD

**Issue by linktohack** [ReactiveX/RxKotlin#110]
This seems to have attracted some discussions here, https://www.reddit.com/r/androiddev/comments/5a0kr6/rxjava_20_final_has_been_released/

What is the Kotlin's idiomatic way to handle null value in an Observable?

**Comment by stepango** [comment]
I personally prefer to use Optional, for Android you could try kOptional or funKTionale's Option.

**Comment by thomasnield** [comment]
Yeah a few weeks back we discussed this at length in the #rx channel on Kotlin Slack. The conclusion we came to is not to force upon people a specific `Optional` implementation but rather just let people choose their own and create the wanted extension functions on their own domain. 

AFAIK, there is no idiomatic way to handle Optionals in RxKotlin that is worth standardizing until 1) Java 8+ is used everywhere or 2) RxJava is completely rewritten from scratch in Kotlin, which I don't think is going to happen.
